merkledag/node.go

```diff
@@ -76,6 +76,11 @@ func MakeLink(n *Node) (*Link, error) {
 	}, nil
 }
 
+// GetCachedNode returns the MDAG Node that was cached, or nil
+func (l *Link) GetCachedNode() *Node {
```

Comment:
Are you really sure of that? In this other issue, [you said](https://github.com/ipfs/go-ipld/issues/14#issuecomment-152260693)

> The 'Node' field in the link objects is very useful, but isnt something we want serialized, its just a cache to make working on the trees easier. Removing it would be fine, but might make certain operations on dags a little slower.

See also ipfs/go-ipld#17, we don't need to remove this pointer.